src/main/java/com/google/sps/data/SubjectCoursesCriteria.java

```diff
@@ -0,0 +1,30 @@
+package com.google.sps.data;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+public class SubjectCoursesCriteria implements Preference {
+  public String preferredSubject;
+
+  public SubjectCoursesCriteria(String preferredSubject) {
+    this.preferredSubject = preferredSubject;
+  }
+
+  /**
+   * Implements the preference score of the schedule, based on the criteria
+   * that prioritizes the number of courses pertaining to a particular subject
+   */
+  public float preferenceScore(Schedule schedule) {
+    int preferredSubjectCount = 0;
+    List<Course> courses = (List) schedule.getCourses();
+    for (Course course : courses) {
```

Comment:
nit: it's considered more legible to use a linebreak before and after code blocks if (and only if) you have other lines of code before and/or after it.

Here are all four possible cases and how they should be broken out with newlines:

```
int foo() {
  while (something) {
    doSomethingElse();
  }
}

int foo() {
  int another = lineOfCode;

  while (something) {
    doSomethingElse();
  }
}

int foo() {
  while (something) {
    doSomethingElse();
  }

  int another = lineOfCode;
}

int foo() {
  int first = lineOfCode;

  while (something) {
    doSomethingElse();
  }

  int last = lineOfCode;
}
```

This applies for any nested code blocks, so... `if/else`, `switch`, `while` and so on would most often be written like this.

Reply:
Sounds good. I made the changes to the 3 criteria files